src/arr/trove/statistics.arr

```diff
@@ -204,3 +230,62 @@ fun r-squared(x :: List<Number>, y :: List<Number>, f :: (Number -> Number)) ->
   end
 end
 
+fun t-test-paired(l1 :: List, l2 :: List) -> Number:
+  doc: "t-test-paired"
+  n1 = l1.length()
+  n2 = l2.length()
+  if n1 <> n2:
+    raise(E.message-exception("t-test-paired: input lists must have equal lengths"))
+  else if n1 == 0:
+    raise(E.message-exception("t-test-paired: input lists must have at least one element"))
+  else:
+    diffs = map2(lam(x1, x2): x1 - x2 end, l1, l2)
+    diffs-mean = mean(diffs)
+    s-hat = stdev-sample(diffs)
+    diffs-mean / (s-hat / num-sqrt(n1))
+  end
+end
+
+fun t-test-pooled(l1 :: List, l2 :: List) -> Number:
+  doc: "t-test-pooled"
+  n1 = l1.length()
+  n2 = l2.length()
+  if (n1 == 0) or (n2 == 0):
+    raise(E.message-exception("t-test-pooled: input lists must have at least one element"))
+  else:
+    m1 = mean(l1)
+    m2 = mean(l2)
+    v1 = variance-sample(l1)
+    v2 = variance-sample(l2)
+    (m1 - m2) / (((((n1 - 1) * num-expt(v1, 2)) + ((n2 - 1) * num-expt(v2, 2))) / ((n1 + n2) - 2)) * num-sqrt((1 / n1) + (1 / n2)))
```

Comment:
This formula is really hard to read.  I'd write either `num-sqr(...)` or `(... * ...)` everywhere you're using `num-expt(..., 2)`, and I'd add more local variables and/or line-breaks to read this; I can't visually tell if this formula is "right" or not.